RedditSharp/Reddit.cs

```diff
@@ -288,6 +289,18 @@ public JToken GetToken(Uri uri)
             var json = JToken.Parse(data);
 
             return json[0]["data"]["children"].First;
+        }
+
+        public Post GetPost(string subreddit, string linkId)
+        {
+            if (linkId.IndexOf("_", StringComparison.Ordinal) > 0)
+            {
+                linkId = linkId.Substring(3, linkId.Length - 3);
+            }
+            var basePostUrl = string.Format(GetPostUrlBase, subreddit, linkId);
+            var url = string.Format(GetPostUrl, basePostUrl);
+            var get = WebAgent.CreateGet(url);
+            return GetPost(get.RequestUri);

```

Comment:
Can you refactor this to avoid calling CreateGet just to get the Uri, and CreatePost again later?